src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/cleanup/CleanUpJobService.java

```diff
@@ -0,0 +1,193 @@
+package uk.gov.hmcts.reform.wataskmonitor.services.jobs.cleanup;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.wataskmonitor.clients.CamundaClient;
+import uk.gov.hmcts.reform.wataskmonitor.config.job.CleanUpJobConfig;
+import uk.gov.hmcts.reform.wataskmonitor.domain.camunda.HistoricCamundaTask;
+import uk.gov.hmcts.reform.wataskmonitor.domain.jobs.GenericJobOutcome;
+import uk.gov.hmcts.reform.wataskmonitor.domain.jobs.GenericJobReport;
+import uk.gov.hmcts.reform.wataskmonitor.services.ResourceEnum;
+import uk.gov.hmcts.reform.wataskmonitor.utils.ResourceUtility;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyList;
+import static uk.gov.hmcts.reform.wataskmonitor.domain.taskmonitor.JobName.TASK_CLEAN_UP;
+import static uk.gov.hmcts.reform.wataskmonitor.services.ResourceEnum.ACTIVE_PROCESS_DELETE_REQUEST;
+import static uk.gov.hmcts.reform.wataskmonitor.services.ResourceEnum.CAMUNDA_CLEAN_UP_TASK_QUERY;
+import static uk.gov.hmcts.reform.wataskmonitor.services.ResourceEnum.HISTORIC_PROCESS_DELETE_REQUEST;
+
+@Component
+@Slf4j
+@SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+public class CleanUpJobService {
+
+    public static final String CAMUNDA_DATE_TIME_REQUEST_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+    public final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(CAMUNDA_DATE_TIME_REQUEST_PATTERN);
+
+    private final CamundaClient camundaClient;
+    private final CleanUpJobConfig cleanUpJobConfig;
+
+    @Autowired
+    public CleanUpJobService(CamundaClient camundaClient,
+                             CleanUpJobConfig cleanUpJobConfig) {
+        this.camundaClient = camundaClient;
+        this.cleanUpJobConfig = cleanUpJobConfig;
+    }
+
+    public boolean isAllowedEnvironment() {
+        log.info("{} cleanUpJobConfig: {}", TASK_CLEAN_UP.name(), cleanUpJobConfig);
+
+        if (!cleanUpJobConfig.getAllowedEnvironment()
+            .contains(cleanUpJobConfig.getEnvironment().toLowerCase(Locale.ROOT))) {
+            log.info("{} is not enabled for this environment: {}",
+                TASK_CLEAN_UP.name(), cleanUpJobConfig.getEnvironment());
+            return false;
+        }
+
+        log.info("{} is enabled for this environment: {}", TASK_CLEAN_UP.name(), cleanUpJobConfig.getEnvironment());
+        return true;
+    }
+
+    public List<HistoricCamundaTask> retrieveProcesses() {
+        List<HistoricCamundaTask> historyTasks = camundaClient.getHistoryProcesses(
```

Comment:
You could call `isAllowedEnvironment` as the first line in here to prevent running in Prod by mistake.  Someone could write new code that calls this method directly without checking.  We don't ever want this to run in Prod.

Reply:
In fact you can call that method before any method that may execute a deletion.